src/wasm/wasm-emscripten.cpp

```diff
@@ -444,6 +475,106 @@ void EmscriptenGlueGenerator::replaceStackPointerGlobal() {
   wasm.removeGlobal(stackPointer->name);
 }
 
+struct StackLimitEnforcer : public PostWalker<StackLimitEnforcer> {
```

Comment:
Walkers are not parallel, you need to use a Pass to get parallelization. Here is an example: https://github.com/WebAssembly/binaryen/pull/2245/files#diff-93cc711d9a969872e9d53bbfe159b92eR129